pkg/xds/generator/outbound_proxy_generator.go

```diff
@@ -339,12 +364,16 @@ func (OutboundProxyGenerator) determineRoutes(
 				envoy_common.WithService(service),
 				envoy_common.WithName(name),
 				envoy_common.WithWeight(destination.GetWeight().GetValue()),
-				envoy_common.WithTags(destination.Destination),
+				envoy_common.WithTags(envoy_common.Tags(destination.Destination).WithoutTags("kuma.io/mesh")),
```

Comment:
It's only a way to signify a cross mesh endpoint, it's not actually a tag for the service? When mesh is its own field on destination this should become irrelevant

Reply:
@jakubdyszkiewicz @lahabana How much do we care about making `mesh` its own field? I'm more and more tempted to smuggle it through everywhere as a tag, because including it as a field means changing some protos (`Dataplane` and `TrafficRoute_Split`) and it's not something users should be trying to set in general.

Reply:
Ok, let's stay with the tag for now, we can convert it to `mesh` field later if needed.

Please explain in the comment why we are removing the tag.